consensus/core/src/dag_state.rs

```diff
@@ -970,6 +1081,59 @@ impl DagState {
     }
 }
 
+struct BlockInfo {
+    block: VerifiedBlock,
+    // Accumulates implicit accept votes for all transactions.
+    accept_votes: StakeAggregator<QuorumThreshold>,
+    // Accumulates reject votes per transaction.
+    reject_votes: BTreeMap<TransactionIndex, StakeAggregator<QuorumThreshold>>,
```

Comment:
Left it as a comment on the DagState as well, but we might want to do some validation on the indexes we insert here as votes.